core/Application/EnvironmentManipulator.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Application;
+
+use Piwik\Application\Kernel\GlobalSettingsProvider;
+
+/**
+ * Used to manipulate Environment instances before the container is created.
+ * Only used by the testing environment setup code, shouldn't be used anywhere
+ * else.
+ */
+interface EnvironmentManipulator
+{
+    /**
+     * Create a custom GlobalSettingsProvider kernel object, overriding the default behavior.
+     *
+     * @param array $kernelObjects
+     * @return GlobalSettingsProvider
+     */
+    public function makeGlobalSettingsProvider(array $kernelObjects);
```

Comment:
a reason not to have this signature:

```php
public function makeGlobalSettingsProvider(GlobalSettingsProvider $provider, PluginList $pluginList)
```

Do we expect the signature to change in the future? And even then it will be easy to refactor (we don't expect 3rd party plugins to implement that right?)